File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Stop.java
package com.kieran.winnipegbusbackend.TripPlanner.classes;

import org.json.JSONException;
import org.json.JSONObject;

public class Stop extends Location {
    int stopNumber;

    public Stop(JSONObject location) {
        super(location);

        try {
            stopNumber = location.getInt("key");
            title = location.getString("name");
        } catch (JSONException e) {

        }
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Stop.kt
package com.kieran.winnipegbusbackend.TripPlanner.classes

import org.json.JSONException
import org.json.JSONObject

class Stop(location: JSONObject) : Location(location) {
    internal var stopNumber: Int = 0

    init {

        try {
            stopNumber = location.getInt("key")
            title = location.getString("name")
        } catch (e: JSONException) {

        }

    }
}
